pandas/core/nanops.py

```diff
@@ -1362,6 +1362,14 @@ def _nanpercentile_1d(values, mask, q, na_value, interpolation):
     quantiles : scalar or array
     """
     # mask is Union[ExtensionArray, ndarray]
+    if values.dtype.kind == "m":
+        # need to cast to integer to avoid rounding errors in numpy
+        result = _nanpercentile_1d(values.view("i8"), mask, q, na_value, interpolation)
+
+        # Note: we have to do do `astype` and not view because in general we
+        #  have float result at this point, not i8
+        return result.astype(values.dtype)
```

Comment:
why are you not letting this then fall thru, e.g. if there is masking here?

Reply:
If we just return `result` here, at this point it is in general float64.  In the calling function we're going to do a `result.view(values.dtype)`, which will be incorrect

Reply:
right my point is to do this *later* in the function after masking and checking for 0-len.

Reply:
just tried that, breaks because it ends up double-masking

Reply:
i wouldnt hang up on this too much, since we're going to have to move this to earlier in the call stack in #27428

Reply:
ok, if  you can clean this up later would be great.